### DISCUSSION TITLE
ZBBridge: Support for Nous A1Z Smart Socket PowerPlug

### DISCUSSION THREAD

**Discussion by rantwolf** [arendst/Tasmota#17498] [Q&A]
### PROBLEM DESCRIPTION
I bought a couple of these Zigbee devices "Nous A1Z Smart Socket".
Now I saw the support for this device is missing in the Sonoff ZigbeeBridge Pro/tasmota firmware.
How can I help to improve missing support?
Found something that the project zigbee2mqtt has support for the power plug.
The console output shows this after pairing: `tele/zbb1_354F4C/SENSOR = {"ZbReceived":{"0xB4A2":{"Device":"0xB4A2","Power":1,"Endpoint":1,"LinkQuality":116}}}`

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff ZigbeeBridge Pro
- [x] Tasmota binary firmware version number used: 12.3.1
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: esptool
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
18:15:26.258 MQT: stat/zbb1_354F4C/RESULT = {"NAME":"Sonoff Zigbee Pro","GPIO":[0,0,576,0,480,0,0,0,0,1,1,5792,0,0,0,3552,0,320,5793,3584,0,640,608,32,0,0,0,0,0,1,0,0,0,0,0,0],"FLAG":0,"BASE":1}
18:15:26.481 MQT: stat/zbb1_354F4C/RESULT = {"Module":{"0":"Sonoff Zigbee Pro"}}
18:15:26.735 MQT: stat/zbb1_354F4C/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"576":"LedLink_i"},"GPIO3":{"0":"None"},"GPIO4":{"480":"Buzzer"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"5792":"Zigbee Rst1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"3552":"Zigbee Tx"},"GPIO20":{"0":"None"},"GPIO21":{"320":"Led_i1"},"GPIO22":{"5793":"Zigbee Rst2"},"GPIO23":{"3584":"Zigbee Rx"},"GPIO24":{"0":"None"},"GPIO25":{"640":"I2C SDA1"},"GPIO26":{"608":"I2C SCL1"},"GPIO27":{"32":"Button1"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
18:02:31.692 MQT: stat/zbb1_354F4C/STATUS = {"Status":{"Module":0,"DeviceName":"sonoff-zbbridge-p1","FriendlyName":["sonoff-zbbridge-p1"],"Topic":"zbb1_354F4C","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
18:02:31.703 MQT: stat/zbb1_354F4C/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"https://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Vbat power on reset","Uptime":"2T21:21:31","StartupUTC":"2022-12-20T19:41:00","Sleep":50,"CfgHolder":4617,"BootCount":10,"BCResetTime":"2022-12-20T20:16:24","SaveCount":20}}
18:02:31.713 MQT: stat/zbb1_354F4C/STATUS2 = {"StatusFWR":{"Version":"12.3.1(zbbrdgpro)","BuildDateTime":"2022-12-16T10:30:45","Core":"2_0_5_3","SDK":"v4.4.3","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 rev.3","CR":"424/699"}}
18:02:31.725 MQT: stat/zbb1_354F4C/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["iot",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
18:02:31.740 MQT: stat/zbb1_354F4C/STATUS4 = {"StatusMEM":{"ProgramSize":1412,"Free":635,"Heap":123,"StackLowMark":4,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DIO","Features":["00000809","8F8A87CF","00548001","000000CF","010013C0","C0000981","40024080","00200000","5400082C","00000080"],"Drivers":"1,2,3,4,5,8,9,10,12,16,20,21,23,24,27,29,35,38,41,50,52,59,62,82","Sensors":"1,2,3,5,6,127"}}
18:02:31.750 MQT: stat/zbb1_354F4C/STATUS5 = {"StatusNET":{"Hostname":"zbb1-354F4C-3916","IPAddress":"192.168.34.30","Gateway":"192.168.34.1","Subnetmask":"255.255.255.128","DNSServer1":"192.168.34.1","DNSServer2":"0.0.0.0","Mac":"XX:XX:XX:XX:XX:XX","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"0.0.0.0","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00"},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
18:02:31.760 MQT: stat/zbb1_354F4C/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.34.120","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_614F4C","MqttUser":"mqtt","MqttCount":2,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
18:02:31.770 MQT: stat/zbb1_354F4C/STATUS7 = {"StatusTIM":{"UTC":"2022-12-23T17:02:31","Local":"2022-12-23T18:02:31","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"08:41","Sunset":"16:56"}}
18:02:31.783 MQT: stat/zbb1_354F4C/STATUS10 = {"StatusSNS":{"Time":"2022-12-23T18:02:31"}}
18:02:31.797 MQT: stat/zbb1_354F4C/STATUS11 = {"StatusSTS":{"Time":"2022-12-23T18:02:31","Uptime":"2T21:21:31","UptimeSec":249691,"Heap":121,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":2,"Berry":{"HeapUsed":21,"Objects":307},"Wifi":{"AP":1,"SSId":"iot","BSSId":"XX:XX:XX:XX:XX:XX","Channel":9,"Mode":"11n","RSSI":100,"Signal":-49,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Why do you say that support is "missing"? This looks like it is supported:
`tele/zbb1_354F4C/SENSOR = {"ZbReceived":{"0xB4A2":{"Device":"0xB4A2","Power":1,"Endpoint":1,"LinkQuality":116}}}`
You might want to consider using `ZbName` and other settings for a more friendly format.

**Comment by rantwolf** [comment]
Yes, it was my fault.
Sorry these are my first steps with zigbee and Tasmota.
I said there is no support but for now I found a workaround.
The problem was that the smart plug does report all cluster information automatically.

Last week I visited a friend and he has a deconz conbee stick.
With this stick and this github link [https://github.com/dresden-elektronik/deconz-rest-plugin/issues/6594](url) I could read all available cluster information about the A1Z smart plug.
Now I created a rule to read all needed information and publish it via mqtt.

One problem I see is also that the smart plug doesn't say: "I'm a router."
The only way to see the meshing function is in the "zigbee map".

But the main question is open...
Is there a way to get all available cluster information from a connected device/router in one query?
I do not have such a conbee stick to get all the information.

PS: Yes, I changed the device name to a better readable vaule via ZBSend, this makes it so easy.

**Reply by Chrisse68** [reply]
Any chance to get the rule you wrote or  short explanation how to obtain it. I am also new to Tasmota and the ZigBee bridge. And yes, I also bought those frickin Smart plugs :)
